src/Common/tests/System/Collections/IDictionary.Generic.Tests.cs

```diff
@@ -323,23 +341,34 @@ public void IDictionary_Generic_Keys_ModifyingTheDictionaryUpdatesTheCollection(
         {
             IDictionary<TKey, TValue> dictionary = GenericIDictionaryFactory(count);
             ICollection<TKey> keys = dictionary.Keys;
+            int previousCount = keys.Count;
             if (count > 0)
                 Assert.NotEmpty(keys);
             dictionary.Clear();
-            Assert.Empty(keys);
+            if (IDictionary_Generic_Keys_Values_ModifyingTheDictionaryUpdatesTheCollection)
+            {
+                Assert.Empty(keys);
+            }
+            else
+            {
+                Assert.Equal(previousCount, keys.Count);
+            }
         }
 
         [Theory]
         [MemberData(nameof(ValidCollectionSizes))]
         public void IDictionary_Generic_Keys_Enumeration_ParentDictionaryModifiedInvalidates(int count)
         {
-            IDictionary<TKey, TValue> dictionary = GenericIDictionaryFactory(count);
-            ICollection<TKey> keys = dictionary.Keys;
-            IEnumerator<TKey> keysEnum = keys.GetEnumerator();
-            dictionary.Add(GetNewKey(dictionary), CreateTValue(3432));
-            Assert.Throws<InvalidOperationException>(() => keysEnum.MoveNext());
-            Assert.Throws<InvalidOperationException>(() => keysEnum.Reset());
-            var cur = keysEnum.Current;
+            if (IDictionary_Generic_Keys_Values_Enumeration_ThrowsInvalidOperation_WhenParentModified)
```

Comment:
I sure wish we could use ConditionalFacts/ConditionalTheories on non-static methods.